core/src/ldml/ldml_processor.cpp

```diff
@@ -390,14 +394,14 @@ ldml_event_state::emit_difference(const std::u32string &old_ctxt, const std::u32
   // marker change.
   // We can detect this because the unchanged_prefix will end with u+FFFF U+0008
   //
-  // Oh, and yes, test case 'regex-test-8a-0' hits this.
+  // k_212* and k_213* hit this case.
   std::u32string common_prefix(old_ctxt.begin(), ctxt_prefix.first);
   if (common_prefix.length() >= 2) {
     auto iter = common_prefix.rbegin();
     if (*(iter++) == LDML_MARKER_CODE && *(iter++) == UC_SENTINEL) {
       // adjust the iterator so that the "U+FFFF U+0008" is not a part of the common prefix.
-      ctxt_prefix.first -= 2;
-      ctxt_prefix.second += 2;
+      ctxt_prefix.first -= 2;  // backup the 'mismatch' point to before the FFFF
+      ctxt_prefix.second -= 2; // backup the 'mismatch' point to before the FFFF
```

Comment:
this was the bug here. I think I switched the sign sometime between the initial fix and commit. Coverage would have helped here.